pkg/monitor/datapath_sock_trace.go

```diff
@@ -31,3 +40,49 @@ type TraceSockNotify struct {
 	L4Proto    uint8
 	Flags      uint8
 }
+
+func (t *TraceSockNotify) IsIpv6() bool {
+	return (t.Flags & TraceSockNotifyFlagIPv6) != 0
+}
+
+func (t *TraceSockNotify) DumpDebug(prefix string) {
+	buf := bufio.NewWriter(os.Stdout)
+
+	fmt.Fprintf(buf, "%s [%s] cgroup_id: %d sock_cookie: %d, dst [%s:%d] %s \n",
```

Comment:
nit: if you read the IP / port into a `netip.AddrPort` then you can get consistent addr+port formatting for IPv4/IPv6: https://pkg.go.dev/net/netip#AddrPort

Reply:
I suppose we can replace `net.IP` with `netIP.Addr`, and then use the `AddrPort` type you mentioned.

Reply:
Yeah unfortunately the datapath format would be different. This would be primarily just to get the default formatting from AddrPort. Main reason was that I noticed in the IPv4 output it was rendered like `[172.20.0.178:80]` and it's unusual to use square brackets for that case. But then I know that the  `[]` are common for IPv6 addresses. The other thing I wasn't sure is that I  know IPv6 addresses can have `:` in them, so using `:` as a delimiter for the port is confusing and I couldn't recall what the typical syntax for that is.

Not a big deal, I just figured if we could assemble the type here & use it purely for the formatter then that might solve these questions.

Reply:
I actually think it would be helpful to use addrport here, though the alternative would be to just open-code their string function: https://cs.opensource.google/go/go/+/refs/tags/go1.19.1:src/net/netip/netip.go;l=1144

Reply:
^ In particular, for IPv6 the `[]` surround the _IP_ and not the port, so then it disambiguates the port delimeter. `[f000::ba12]:1234`.

Reply:
We can surround the IP with [...] in `fmt.Fprintf`.